controllers/secretscope_controller_databricks.go

```diff
@@ -152,25 +152,66 @@ func (r *SecretScopeReconciler) submitACLs(instance *databricksv1alpha1.SecretSc
 	return nil
 }
 
-func (r *SecretScopeReconciler) submit(instance *databricksv1alpha1.SecretScope) error {
+// checkCluster checks if Databricks cluster supports ACLs, and checks if secret scope exists.
+func (r *SecretScopeReconciler) checkCluster(instance *databricksv1alpha1.SecretScope) error {
 	scope := instance.ObjectMeta.Name
 	initialManagePrincipal := instance.Spec.InitialManagePrincipal
 
+	// try create secret scope to see if exists or not.
 	err := r.APIClient.Secrets().CreateSecretScope(scope, initialManagePrincipal)
```

Comment:
Can we use get instead of creating? or can we use https://docs.azuredatabricks.net/dev-tools/api/latest/secrets.html#get-secret-acl?

Reply:
I don't think we can use the _get secret acl_ approach. If the user isn't using a premium tier workspace, the user will get back `Error: {"error_code":"PERMISSION_DENIED","message":"ACL is not supported in your workspace."}`.

However, we could probably list the secret scopes and filter the one we want. That might work.

Reply:
Update: Listing the SecretScopes is fine if we're just checking if the SecretScope exists or not but when it comes to checking if ACLs is supported or not, if the SecretScope doesn't exist prior to checking, a `RESOURCE DOES NOT EXIST` is returned rather than a `ACL is not supported in your workspace.` which is what we'd want to see. I've removed ACL checks for now.

Reply:
I've added back in ACL checks but I've cleaned up the implementation a little.